src/modules/storage/LedgerStorage.ts

```diff
@@ -2188,4 +2198,95 @@ export class LedgerStorage extends Storages
+                                              FROM
+                                              tx_outputs
+                                              where address = ?;`  
+                 let freeze_sql   =        `SELECT 
+                                               SUM(amount) as freeze_amount
+                                               FROM 
+                                               utxos
+                                               WHERE 
+                                               type = 1 and address = ?;`
+
+                this.query(balance_sql,[address!.toString()])
+                .then(async(amount:any)=>{
+                    balance = amount[0].amount;
+                   return this.query(tx_count_sql,[address!.toString()]);
+                }).then((count: any)=>{
+                   txs_count = count[0].tx_count;
+                   return this.query(freeze_sql,[address!.toString()])
+                }).then((freeze: any)=>{
+                    freeze_amount = freeze[0].freeze_amount;                    
+                   let res = {
+                       balance,
+                       txs_count,
+                       freeze_amount,
+                       address,
+                       received_rewards:468768,
+                       precentage:'45%',
+                       value:'$8468'
 }
+                   resolve(res);
+                  });
+            })();
+        });
+}
+      /**
+     * Get Adresss, Balance, Freeze Amount
+     * @returns returns the Promise with requested data
+     * and if an error occurs the `.catch` is called with an error. 
+     */
+       public getBOAHolders(limit: number, page: number): Promise<any>{
+        return new Promise<any>((resolve, reject) =>
+        {
+            (async () =>    
+            {   
+                let data:any=[];
+                let freeze_amount = JSBI.BigInt(0);
+                let balance = JSBI.BigInt(0);
+                let boaHolder_sql  =       `SELECT address, SUM(amount) as balance,
+                                              (SELECT SUM(amount) from utxos O where O.type = '1' and O.address = U.address) as freeze
+                                              FROM utxos U GROUP BY address
+                                              LIMIT ? OFFSET ?;`
+                                              
+                this.query(boaHolder_sql,[limit, limit * (page - 1)])
+                .then(async(rows:any)=>{
+
+                    for(let i=0;i< rows.length;i++){
+                        let result= {
+                            address:rows[i].address,
+                            balance: rows[i].balance,
+                            freeze_amount:rows[i].freeze,
+                            received_rewards:468768,
+                            precentage:'45%',
+                            value:'$8468'
+                        }
+                        data.push(result)
+                    }
```

Comment:
This is the same opinion as above.
Also, it should include sorting functions.(balance, feezing, rewards)